aten/src/ATen/function_wrapper.py

```diff
@@ -852,6 +891,20 @@ def format_return_type(return_types):
             return return_types[0]['type']
         return "std::tuple<{}>".format(','.join(r['type'] for r in return_types))
 
+    def process_schema_order_actual(schema_order_actual):
+        if schema_order_actual == 'dtype':
+            return 'optTypeMetaToScalarType(options.dtype_opt())'
+        elif schema_order_actual == 'layout':
+            return 'options.layout_opt()'
+        elif schema_order_actual == 'device':
+            return 'options.device_opt()'
+        elif schema_order_actual == 'pin_memory':
+            return 'options.pinned_memory_opt()'
+        elif schema_order_actual == 'memory_format':
+            return 'c10::impl::process_memory_format(options, memory_format)'
+        else:
+            return schema_order_actual
```

Comment:
The correctness of this translation CRITICALLY relies on the defaulting of these arguments to be `None`, and this defaulting is not universally true:

```
- func: tril_indices(int row, int col, int offset=0, *, ScalarType? dtype=long, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor
  dispatch:
    CPU: tril_indices_cpu
    CUDA: tril_indices_cuda
```

See also:

```
- func: _empty_per_channel_affine_quantized(int[] size, *, Tensor scales, Tensor zero_points, int axis, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None, MemoryFormat? memory_format=contiguous_format) -> Tensor
  category_override: factory
  dispatch:
    CPU: empty_per_channel_affine_quantized_other_backends_stub
    QuantizedCPU: empty_per_channel_affine_quantized_cpu
```

@bhosmer and I attempted to fix the MemoryFormat situation in https://github.com/pytorch/pytorch/pull/34251 but IIRC it was not so easy because the dtype defaulting choices here affect how we do Python argument parsing, and presence of the default perturbs JIT analysis.

Reply:
Actually, it might not be as bad as this implies, because I think we may be handling defaulting in the C++ frontend incorrectly anyway. Current handling for `tril_indices` is:

```
CAFFE2_API Tensor tril_indices(int64_t row, int64_t col, int64_t offset=0, const TensorOptions & options=at::kLong);
```

Which would work OK with the code that you've written here (and will give you a strange result if you try to pass in non-default tensor options, as you'll "lose" the kLong default.)

Reply:
hm yeah this sounds bad. We should probably fix the C++ frontend in a follow up.